client/scripts/DrawTogether.js

```diff
@@ -1295,6 +1326,262 @@ DrawTogether.prototype.handlePaintSelection = function handlePaintSelection (eve
 	}.bind(this));
 };
 
+// Send null for unchanging value
+DrawTogether.prototype.updateIndividualFavoriteDom = function updateIndividualFavoriteDom(newX, newY, newName, newOwner, element) {
+	if(newX !== null)
+		element.dataset.x = newX;
+	if(newY !== null)
+		element.dataset.y = newY;
+	if(newName !== null)
+		element.dataset.name = newName;
+	if(newOwner !== null)
+		element.dataset.owner = newOwner;
+	
+	var coordinateButton = element.getElementsByClassName("fav-coor-button")[0];
+	var inputRename = element.getElementsByClassName("fav-rename-input")[0];
+	if ( (newName || element.dataset.name) === ""){
+		inputRename.value = "";
+		coordinateButton.innerHTML = (newX || element.dataset.x) + "," + (newY || element.dataset.y);
```

Comment:
If you want to add text to an element you have to do it using createTextNode.
This works in all browsers and makes it impossible for users to add html for better security.
It is also a lot faster and cleaner.

while (element.firstChild) element.removeChild(element.firstChild); //To clear the previous text
element.appendChild(document.createTextNode("YOUR TEXT HERE"));